plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -0,0 +1,247 @@
+
+        if(data["DHCPv6"]["UseNTPServers"] is True):
+            payload["DHCPv6"]["UseNTPServers"] = False
+            resDHV6 = self.patch_request(self.root_uri + ethuri, payload)
+            if resDHV6['ret'] is False:
+                return resDHV6
+
+        DTuri = self.manager_uri + "DateTime"
+
+        response = self.get_request(self.root_uri + DTuri)
+        if(response['ret'] is False):
+            return response
+
+        data = response['data']
+
+        NTPlist = data[setkey]
+        if(len(NTPlist) == 2):
+            NTPlist.pop(0)
+
+        NTPlist.append(mgr_attributes['mgr_attr_value'])
+
+        payload = {}
+        payload[setkey] = NTPlist
+
+        response1 = self.patch_request(self.root_uri + DTuri, payload)
+        if response1['ret'] is False:
+            return response1
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % mgr_attributes['mgr_attr_name']}
+
+    def setTimeZone(self, attr):
+        # result = {}
+        key = attr['mgr_attr_name']
+
+        uri = self.manager_uri + "DateTime/"
+        response = self.get_request(self.root_uri + uri)
+        if(response['ret'] is False):
+            return response
+
+        data = response["data"]
+
+        if key not in data:
+            return {'ret': False, 'msg': "Key %s not found" % key}
+
+        TimeZonelist = data["TimeZoneList"]
+        index = ""
+
+        for i in range(0, len(TimeZonelist)):
+            if(attr['mgr_attr_value'] in TimeZonelist[i]["Name"]):
+                index = TimeZonelist[i]["Index"]
+                break
+
+        payload = {key: {"Index": index}}
+        # toset = "{\"Index\":" + str(index) + "}"
+
+        # payload = {key: json.loads(toset)}
+
+        response = self.patch_request(self.root_uri + uri, payload)
+        if(response['ret'] is False):
+            return response
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % attr['mgr_attr_name']}
+
+    def set_DNSserver(self, attr):
+        # result = {}
+        key = attr['mgr_attr_name']
+        # setkey = "Oem"
+
+        nic_info = self.get_manager_ethernet_uri()
+        uri = nic_info["nic_addr"]
+
+        response = self.get_request(self.root_uri + uri)
+        if(response['ret'] is False):
+            return response
+
+        data = response['data']
+
+        DNSlist = data["Oem"]["Hpe"]["IPv4"][key]
+
+        if(len(DNSlist) == 3):
+            DNSlist.pop(0)
+
+        DNSlist.append(attr['mgr_attr_value'])
+
+        payload = {
+            "Oem": {
+                "Hpe": {
+                    "IPv4": {
+                        key: DNSlist
+                    }
+                }
+            }
+        }
+
+        response = self.patch_request(self.root_uri + uri, payload)
+        if(response['ret'] is False):
+            return response
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % attr['mgr_attr_name']}
+
+    def set_DomainName(self, attr):
```

Comment:
The naming convention here is a bit odd: it is neither `CamelCase` nor `snake_case`. Python convention is to use `snake_case` for methods, functions and variables, and `CamelCase` for classes.

While I strongly recommend using `snake_case`, because it is the language's convention, you are free to use `CamelCase` if that makes more sense to you - even if it won't to most python programmers that come and read your code later. 

However, at the very least we **MUST** have some consistency. This mixed style is not usual and it makes reading the code a bit more uncomfortable than what it should be.

Reply:
still some odd cases floating around, but thanks for fixing the others